index/index.go

```diff
@@ -230,6 +234,11 @@ type FieldDict interface {
 	Close() error
 }
 
+type AdvFieldDict interface {
+	FieldDict
+	Exists(key []byte) (bool, error)
+}
+
```

Comment:
One thought... instead of having AdvFieldDict extend the FieldDict interface, maybe this new interface can just have the Exists() method all by itself?  Perhaps that might help clean up the code, if the interface-casting check can then happen in simpler ways?  e.g., wondering if upsidedown's dictionary iterators just don't implement the Exists() method, and only scorch dictionary iterators have that new method?

Related, there seem to be a few synonyms for this new fast-path approach... "Random" access lookup of a term, "Adv"anced lookup, or whether a key or term "Exists" in the dictionary.  Would recommend instead just going with just a single concept name.  If you've no strong preference, I think I'd prefer Exists or (perhaps even KeyExists or TermExists if "Exists" by itself is too generic).

Reply:
good catch, thanks. done!